common/src/main/scala/net/psforever/types/SquadRequestType.scala

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2019 PSForever
+package net.psforever.types
+
+import net.psforever.packet.PacketHelpers
+import scodec.codecs._
+
+object SquadRequestType extends Enumeration {
+  type Type = Value
+  val
+  Invite,
+  ProximityInvite,
+  Accept,
+  Reject,
+  Cancel,
+  Leave,
+  Promote,
+  Disband,
+  PlatoonInvite,
+  PlatoonAccept,
+  PlatoonReject,
+  PlatoonCancel,
+  PlatoonLeave,
+  PlatoonDisband
+  = Value
+
+  implicit val codec = PacketHelpers.createEnumerationCodec(this, uint4L)
+
+  def toResponse(request : SquadRequestType.Value) : SquadResponseType.Value = {
+    val id = request.id
+    if(id < 6) SquadResponseType(id)
+    else if(id > 6) SquadResponseType(id - 1)
+    else throw new NoSuchElementException("request does not have an applicable response")
```

Comment:
wtf planetside. can you explain what is happening here?

Reply:
Since I sidestepped this idea at some point, I can probably strip this code out.  It's like this:

When I first inheritted this code from aphedox, he used a single enum for both `Request` and `Response`, though he tied them to the former (as `Request` types).  When I noted that the `Response` packets produced messages in the events chat, I decided to add them to the comments so it would be eaiser to remember when to use them.  At some point, I noticed that for certain enum values the message for value A would be better suited for value A+1 and that's when I decided to create two enumerations.  This code would just make `Request.Invite` translateable to `Response.Invite` if one had a different id than the other.

Reply:
Removed.